packages/mui-joy/src/AltDateWidget/AltDateWidget.tsx

```diff
@@ -0,0 +1,274 @@
+import { MouseEvent, memo, useCallback, useEffect, useReducer, useState } from 'react';
+import {
+  ariaDescribedByIds,
+  parseDateString,
+  toDateString,
+  pad,
+  DateObject,
+  FormContextType,
+  RJSFSchema,
+  StrictRJSFSchema,
+  TranslatableString,
+  WidgetProps,
+} from '@rjsf/utils';
+import Stack from '@mui/joy/Stack';
+import Button from '@mui/joy/Button';
+
+function rangeOptions(start: number, stop: number) {
+  const options = [];
+  for (let i = start; i <= stop; i++) {
+    options.push({ value: i, label: pad(i, 2) });
+  }
+  return options;
+}
+
+function readyForChange(state: DateObject) {
+  return Object.values(state).every((value) => value !== -1);
+}
+
+function dateElementProps(
+  state: DateObject,
+  time: boolean,
+  yearsRange: [number, number] = [1900, new Date().getFullYear() + 2]
+) {
+  const { year, month, day, hour, minute, second } = state;
+  const data = [
+    {
+      type: 'year',
+      range: yearsRange,
+      value: year,
+    },
+    { type: 'month', range: [1, 12], value: month },
+    { type: 'day', range: [1, 31], value: day },
+  ] as { type: string; range: [number, number]; value: number | undefined }[];
+  if (time) {
+    data.push(
+      { type: 'hour', range: [0, 23], value: hour },
+      { type: 'minute', range: [0, 59], value: minute },
+      { type: 'second', range: [0, 59], value: second }
+    );
+  }
+  return data;
+}
+
+type DateElementProps<T = any, S extends StrictRJSFSchema = RJSFSchema, F extends FormContextType = any> = Pick<
+  WidgetProps<T, S, F>,
+  'value' | 'name' | 'disabled' | 'readonly' | 'autofocus' | 'registry' | 'onBlur' | 'onFocus'
+> & {
+  rootId: string;
+  select: (property: keyof DateObject, value: any) => void;
+  type: string;
+  range: [number, number];
+};
+
+function DateElement<T = any, S extends StrictRJSFSchema = RJSFSchema, F extends FormContextType = any>({
+  type,
+  range,
+  value,
+  select,
+  rootId,
+  name,
+  disabled,
+  readonly,
+  autofocus,
+  registry,
+  onBlur,
+  onFocus,
+  endDecorator,
+}: DateElementProps<T, S, F> & any) {
+  const id = rootId + '_' + type;
+  const { SelectWidget } = registry.widgets;
+  return (
+    <SelectWidget
+      schema={{ type: 'integer' } as S}
+      id={id}
+      name={name}
+      className='form-control'
+      options={{ enumOptions: rangeOptions(range[0], range[1]) }}
+      placeholder={type}
+      value={value}
+      disabled={disabled}
+      readonly={readonly}
+      autofocus={autofocus}
+      onChange={(value: any) => select(type as keyof DateObject, value)}
+      onBlur={onBlur}
+      onFocus={onFocus}
+      registry={registry}
+      label=''
+      aria-describedby={ariaDescribedByIds<T>(rootId)}
+      endDecorator={endDecorator}
+    />
+  );
+}
+
+/** The `AltDateWidget` is an alternative widget for rendering date properties.
+ * @param props - The `WidgetProps` for this component
```

Comment:
Add a blank line please
```suggestion
/** The `AltDateWidget` is an alternative widget for rendering date properties.
 *
 * @param props - The `WidgetProps` for this component
```